proposals/lip-introduce_bft_module.md

```diff
@@ -0,0 +1,805 @@
+    if length(bftVotes.blockBFTInfos) > 0:
+        nextHeight = bftVotes.blockBFTInfos[0].height + 1
+    else:
+        # only directly after processing the genesis block we have
+        # length(bftVotes.blockBFTInfos) == 0
+        nextHeight = bftVotes.maxHeightPrevoted + 1
+    bftParameters(nextHeight)=parameters
+
+    # update bftVotes.updateActiveValidatorsVoteInfo
+    newActiveValidatorsVoteInfo = []
+    for each validator in validators:
+        if validator.address appears in bftVotes.activeValidatorsVoteInfo:
+            validatorVoteInfo = object in bftVotes.activeValidatorsVoteInfo
+                                with address equal to validator.address
+        else:
+            validatorVoteInfo = object following schema of elements
+                                in bftVotes.activeValidatorsVoteInfo
+            validatorVoteInfo.address = validator.address
+            validatorVoteInfo.minHeightActive = nextHeight
+            validatorVoteInfo.largestHeightPrecommit = nextHeight-1
+        add validatorVoteInfo to newActiveValidatorsVoteInfo
+    sort newActiveValidatorsVoteInfo lexicographically by address
+    bftVotes.activeValidatorsVoteInfo = newActiveValidatorsVoteInfo
+```
+
+### Endpoints for Off-Chain Services
+
+TBD
+
+### Genesis Block Processing
+
+The following two steps are executed as part of the genesis block processing, see the [“Update genesis block schema and processing” LIP][lip-genesis-block-processing] for details.
+
+#### Genesis State Initialization
+
+As part of the genesis state initialization of a genesis block `b`, the BFT module store is initialized as follows:
+
+* The BFT Parameters substore is empty.
+* The BFT Votes substore is initialized as follows:
+    * `bftVotes.maxHeightPrevoted = b.height`,
+    * `bftVotes.maxHeightPrecommitted = b.height`,
+    * `bftVotes.maxHeightCertified = b.height`,
+    * `bftVotes.blockBFTInfos` is an empty array,
+    * `bftVotes.activeValidatorsVoteInfo` is an empty array.
+
+#### Genesis State Finalization
+
+In this step the BFT module does not execute any logic.
+However, the exposed function `setBFTParameters` is expected to be called by the PoA module or DPoS module as part of the genesis state finalization of these modules.
+
+### Block Processing
+
+The following steps are executed as part of the (non-genesis) block processing, see [LIP 0055][lip-block-processing] for details.
+
+#### After Transactions Execution
+
+After the transactions of a block `b` are executed, the following logic is executed:
+
+1. The object `getBlockBFTProperties(b.header)` is added to the beginning of the array `bftVotes.blockBFTInfos`.
+2. If `bftVotes.blockBFTInfos[MAX_LENGTH_BLOCK_BFT_INFOS]` exists, then the corresponding value is removed from the array.
+3. The function `applyPrevotesPrecommits` from the LIP ["Add weights to Lisk-BFT consensus protocol"][lip-weighted-lisk-bft] is executed to update the prevote and precommit weights.
```

Comment:
I just spotted by chance that the function name in the other LIP is `updatePrevotesPrecommits` instead of `applyPrevotesPrecommits`.

Reply:
Fixed, thanks!